devito/dse/rewriters.py

```diff
@@ -0,0 +1,412 @@
+import abc
```

Comment:
Note for reviewers: this file is just the result of merging what was in `backends` before. There also are a few changes, but only to a single routine, marked below